scripts/prepDataForGephi.py

```diff
@@ -0,0 +1,330 @@
+parser.add_argument('--mafname', dest='mafname', metavar = 'MAF variable name', help='name of variable corresponding to minor allele frequency')
+parser.add_argument('--casecountname', dest='ccname', metavar = 'case count variable name', help='name of variable corresponding to case count')
+parser.add_argument('--pvaluename', dest='pvname', metavar = 'name of variable for p-value association', help='name of variable corresponding to p-value of association')
+parser.add_argument('--delim', dest='delim', required=True, metavar = 'delimiter used in input PheWAS datasets', help='delimiter used in input PheWAS data')
+
+args = parser.parse_args()
+
+# Example usage:
+# python prepDataForGephi.py --inputdata "./subset/" 
+# --filteredoutput "./filteredSubset/" 
+# --edgemapoutput "./subsetEdgeMap.csv" 
+# --nodemapinput "./phecodeDefinitions.csv" 
+# --nodemapoutput "./subsetNodeMap.csv" 
+# --mafthreshold "0.01" 
+# --casecountthreshold "200" 
+# --pvaluethreshold "1e-8" 
+# --phecodename "phenotypeID" 
+# --snpname "ID" 
+# --mafname "af" 
+# --casecountname "num_cases" 
+# --pvaluename "pval" 
+# --delim " "
+
+directoryPath = args.inputdata
+
+filteredDirectoryPath = args.filteredout
+
+edgeMapOutputPath = args.edgemapout
+
+nodeMapInputPath = args.nodemapin
+nodeMapOutputPath = args.nodemapout
+
+mafThreshold = args.mafthresh
+caseCountThreshold = args.ccthresh
+pValueThreshold = args.pvalthresh
+
+phecodeColName = args.phecodename
+snpColName = args.snpname
+mafColName = args.mafname
+caseCountColName = args.ccname
+pValueColName = args.pvname
+
+fileDelimiter = args.delim
+
+
+# Figure out where each column is in the input data
+phecodeColNumber = None
+mafColNumber = None
+caseCountColNumber = None
+pValueColNumber = None
+snpColNumber = None
+
+# Get the header row of the first file in our input directory
+firstFileName = os.listdir(directoryPath)[0]
+firstFile = open(directoryPath + firstFileName, 'r')
+firstLine = firstFile.readlines()[0]
+firstLineAsArray = firstLine.split(fileDelimiter)
+
+for i in range(0, len(firstLineAsArray)):
+    if(firstLineAsArray[i] == phecodeColName or firstLineAsArray[i] == phecodeColName + "\n"):
+        phecodeColNumber = i
+    elif(firstLineAsArray[i] == str(mafColName) or firstLineAsArray[i] == str(mafColName) + "\n"):
+        mafColNumber = i
+    elif(firstLineAsArray[i] == str(caseCountColName) or firstLineAsArray[i] == str(caseCountColName) + "\n"):
+        caseCountColNumber = i
+    elif(firstLineAsArray[i] == str(pValueColName) or firstLineAsArray[i] == str(pValueColName) + "\n"):
+        pValueColNumber = i
+    elif(firstLineAsArray[i] == snpColName or firstLineAsArray[i] == snpColName + "\n"):
+        snpColNumber = i
+firstFile.close()
+
+# We need a column corresponding to phenotype name in our data. Force an exit if it can't be found
+if(phecodeColNumber == None):
+    sys.exit('ERROR: Cannot identify column corresponding to phenotype name in input data')
```

Comment:
I had allowed for this in my first implementation yesterday. However, if the user wants to add Associated SNPs to their node map, this will cause issues. The edge map will be made with the file names as phenotype names, but then the node map will become empty b/c the file names wouldn't match with the phenotype names in the node map. Let's discuss this.